src/librbd/io/ObjectRequest.cc

```diff
@@ -714,9 +714,8 @@ int ObjectCompareAndWriteRequest<I>::filter_write_result(int r) const {
     I *image_ctx = this->m_ictx;
 
     // object extent compare mismatch
-    uint64_t offset = -MAX_ERRNO - r;
```

Comment:
Should the `r <= -MAX_ERRNO` condition a few lines up updated too, for strict equality at the very least?  Or perhaps `m_mismatch_offset_store != -1` to be more explicit, assuming it's always guaranteed to be set?

I see that `osd_errc::cmpext_failed` is defined to `MAX_ERRNO`, so it would appear to still work, but if `r < -MAX_ERRNO` is no longer possible, it's just a source of confusion.

Reply:
Done.